dev/com.ibm.ws.security.token.ltpa/src/com/ibm/ws/security/token/ltpa/internal/LTPAToken2.java

```diff
@@ -62,11 +62,29 @@
     private final LTPAPublicKey publicKey;
     private String cipher = null;
 
+    private static String javaVendor = null;
+    private static boolean ibmJVM = false;
+    
+
     static {
         MessageDigest m1 = null, m2 = null;
         try {
-	    m1 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM, LTPAKeyUtil.defaultJCEProvider());
-	    m2 = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM, LTPAKeyUtil.defaultJCEProvider());
+	    if (javaVendor == null) {
+		javaVendor = System.getProperty("java.vendor");
```

Comment:
Consider using the `JavaInfo` API (from kernel.service) for checking the java vendor.

Reply:
Will update the code and run the build again.